ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -172,10 +174,27 @@ public StatementExecutorResponse execute(
           + System.lineSeparator()
           + commandRunnerWarningString);
     }
-    final ConfiguredStatement<?> injected = injectorFactory
-        .apply(executionContext, securityContext.getServiceContext())
-        .inject(statement);
+    final InjectorWithSideEffects injector = InjectorWithSideEffects.wrap(
+            injectorFactory.apply(executionContext, securityContext.getServiceContext()));
+    final ConfiguredStatementWithSideEffects<?> injectedWithSideEffects =
+            injector.injectWithSideEffects(statement);
+    try {
+      return executeInjected(
+            injectedWithSideEffects.getStatement(), statement,
+            executionContext,
+            securityContext);
```

Comment:
I assume that **any** error inside `executeInjected` should result in topic deletion if it was created (apart from `ProducerFencedException` probably - discussed separately).